services/management/ia/channels_service.go

```diff
@@ -17,15 +17,242 @@
+		return nil, errors.Wrap(err, "failed to get notification channels")
+	}
+
+	res := make([]*iav1beta1.Channel, len(channels))
+	for i, channel := range channels {
+		c := &iav1beta1.Channel{
+			ChannelId: channel.ID,
+			Summary:   channel.Summary,
+			Disabled:  channel.Disabled,
+		}
+
+		switch channel.Type {
+		case models.Email:
+			config := channel.EmailConfig
+			c.Channel = &iav1beta1.Channel_EmailConfig{
+				EmailConfig: &iav1beta1.EmailConfig{
+					SendResolved: config.SendResolved,
+					To:           config.To,
+				},
+			}
+		case models.PagerDuty:
+			config := channel.PagerDutyConfig
+			c.Channel = &iav1beta1.Channel_PagerdutyConfig{
+				PagerdutyConfig: &iav1beta1.PagerDutyConfig{
+					SendResolved: config.SendResolved,
+					RoutingKey:   config.RoutingKey,
+					ServiceKey:   config.ServiceKey,
+				},
+			}
+		case models.Slack:
+			config := channel.SlackConfig
+			c.Channel = &iav1beta1.Channel_SlackConfig{
+				SlackConfig: &iav1beta1.SlackConfig{
+					SendResolved: config.SendResolved,
+					Channel:      config.Channel,
+				},
+			}
+		case models.WebHook:
+			config := channel.WebHookConfig
+			c.Channel = &iav1beta1.Channel_WebhookConfig{
+				WebhookConfig: &iav1beta1.WebhookConfig{
+					SendResolved: config.SendResolved,
+					Url:          config.URL,
+					HttpConfig:   convertModelToHTTPConfig(config.HTTPConfig),
+					MaxAlerts:    config.MaxAlerts,
+				},
+			}
+		default:
+			return nil, errors.Wrapf(err, "unknown notification channel type %s", channel.Type)
+		}
+
+		res[i] = c
+	}
+
+	return &iav1beta1.ListChannelsResponse{Channels: res}, nil
+}
+
+// AddChannel adds new notification channel.
+func (s *ChannelsService) AddChannel(ctx context.Context, req *iav1beta1.AddChannelRequest) (*iav1beta1.AddChannelResponse, error) {
+	params := &models.CreateChannelParams{
+		Summary:  req.Summary,
+		Disabled: req.GetDisabled(),
+	}
+
+	if emailConf := req.GetEmailConfig(); emailConf != nil {
+		params.EmailConfig = &models.EmailConfig{
+			SendResolved: emailConf.SendResolved,
+			To:           emailConf.To,
+		}
+	}
+	if pagerDutyConf := req.GetPagerdutyConfig(); pagerDutyConf != nil {
+		params.PagerDutyConfig = &models.PagerDutyConfig{
+			SendResolved: pagerDutyConf.SendResolved,
+			RoutingKey:   pagerDutyConf.RoutingKey,
+			ServiceKey:   pagerDutyConf.ServiceKey,
+		}
+	}
+	if slackConf := req.GetSlackConfig(); slackConf != nil {
+		params.SlackConfig = &models.SlackConfig{
+			SendResolved: slackConf.SendResolved,
+			Channel:      slackConf.Channel,
+		}
+	}
+	if webhookConf := req.GetWebhookConfig(); webhookConf != nil {
+		params.WebHookConfig = &models.WebHookConfig{
+			SendResolved: webhookConf.SendResolved,
+			URL:          webhookConf.Url,
+			MaxAlerts:    webhookConf.MaxAlerts,
+			HTTPConfig:   convertHTTPConfigToModel(webhookConf.HttpConfig),
+		}
+	}
+
+	err := s.ia.AddChannel(params)
+	if err != nil {
+		return nil, err
+	}
+
+	return &iav1beta1.AddChannelResponse{}, nil
```

Comment:
@artemgavrilov this needs to be updated to return IDs https://github.com/percona/pmm/pull/628/files#diff-13ddf37687358efd5ba2d959eb9e9ee226db39006b392b9e2ee246b2bf9d80edR127

Reply:
Fixed